### ISSUE TITLE
Clarification for multi-task scenario (gradient conflicts)

### ISSUE THREAD

**Issue by ogencoglu** [Fchaubard/gradient_agreement_filtering#5]
Thanks for the really interesting work.

How does this relate to or work with regarding multi-task (e.g. sum of two loss terms) scenarios where gradient conflicts are common (not only between batches but between loss terms in the same batch). Relevant libraries that try to tackle this problem are:
- https://github.com/tum-pbs/ConFIG
- https://github.com/TorchJD/torchjd

Do you have intuition or results for such cases (most common would be image segmentation where cross entropy loss is usually accompanied by boundary loss/focal loss/dice loss and things of that sort)?

**Comment by ogencoglu** [comment]
And as a separate question: what is the reason for focusing on val accuracy instead of val loss? It would be more interesting to see the same analysis for val loss as that would mean it might generalize to tasks other than classification.

**Comment by Fchaubard** [comment]
Thanks! Sorry for the delay. So the multi-task gradient work is very interesting but distinct from ours, in fact in may even be complimentary if I think about it. If viewing the sequence of batches as a curriculum, where you may (just as an example) round robin through the tasks (one task per batch at a time) then I think these would be complementary as per task you should ensure you are not overfitting to that batch by gradient checking. As for the focus on val accuracy, val accuracy could be replaced with any downstream metric for any task, (bounding box  -> mAP, FID, etc) but its important to note you can improve on val loss but not improve on your goal metric (we can be at 0.9 probability on an image and be right, and see an improve in val loss by pushing to 0.99 probability, improving in CE, but we see no improvement in val acc).
